### PR TITLE
Use connection id instead of channel id in IsMiddlewareEnabled

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR updates  `IsMiddlewareEnabled`, `SetMiddlewareEnabled` and `DeleteMiddlewareEnabled` to use portID and connectionID as opposed to portID and channelID.

closes: #2235

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/ibc_middleware.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ type IBCMiddleware struct {
 	keeper keeper.Keeper
 }
 
-// IBCMiddleware creates a new IBCMiddleware given the associated keeper and underlying application
+// NewIBCMiddleware creates a new IBCMiddleware given the associated keeper and underlying application
```
Comment (chatton) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware.go:26]: unrelated docstring fix

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go [hunk 2]
```diff
@@ -61,6 +62,15 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", fmt.Sprintf("x/%s-%s", host.ModuleName, icatypes.ModuleName))
 }
 
+// GetConnectionID returns the connection id for the given port and channelIDs.
+func (k Keeper) GetConnectionID(ctx sdk.Context, portID, channelID string) (string, error) {
```
Comment (chatton) [inline modules/apps/27-interchain-accounts/controller/keeper/keeper.go:66]: I chose to add this function rather than export the channelKeeper, as so far it's the only time we've needed.

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go [hunk 3]
```diff
@@ -144,11 +154,15 @@ func (k Keeper) GetAllActiveChannels(ctx sdk.Context) []genesistypes.ActiveChann
 	for ; iterator.Valid(); iterator.Next() {
 		keySplit := strings.Split(string(iterator.Key()), "/")
 
+		portID := keySplit[1]
+		connectionID := keySplit[2]
+		channelID := string(iterator.Value())
```
Comment (chatton) [inline modules/apps/27-interchain-accounts/controller/keeper/keeper.go:159]: extracted these out to make it more clear which values are being passed into `IsMiddlewareEnabled`

**Review by damiannolan — APPROVED** [review]
Awesome, LGTM!

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 4]
```diff
@@ -22,12 +22,12 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, connectionID, owner,
...
 	}
 
-	channelID, err := k.registerInterchainAccount(ctx, connectionID, portID, version)
+	_, err = k.registerInterchainAccount(ctx, connectionID, portID, version)
 	if err != nil {
 		return err
 	}
 
-	k.SetMiddlewareEnabled(ctx, portID, channelID)
+	k.SetMiddlewareEnabled(ctx, portID, connectionID)
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/keeper/account.go:30]: Looks like its covered in #2283
